polynote-kernel/src/main/scala/polynote/kernel/ScalaCompiler.scala

```diff
@@ -418,6 +418,26 @@ class ScalaCompiler private (
       exitingTyper(compilationUnit.body)
     }
 
+    private[kernel] def typedTreeAt(pos: Int) = zio.blocking.effectBlocking {
+      val run = new Run()
+      compilationUnit.body = wrapped
+      compilationUnit.lastBody = wrapped
+      unitOfFile.put(sourceFile.file, compilationUnit)
+      // this does a targeted-typecheck, which is more tolerant of errors. But, we have to make sure it's past the
+      // namer first or it will go through the parser which will throw position validation errors.
+      enteringPhase(currentRun.namerPhase)(currentRun.namerPhase.asInstanceOf[GlobalPhase].applyPhase(unitOfFile(sourceFile.file)))
+      compilationUnit.status = JustParsed
+      try {
+        val result = NscThief.typedTreeAt(global, Position.offset(sourceFile, pos))
+        result
+      } catch {
+        case err: Throwable =>
+          val e = err
+          e.printStackTrace()
+          throw e
```

Comment:
How noisy is this? If it's not too noisy I'm all for keeping it

Reply:
Oops I meant to get rid of that

Reply:
Fixed, it was leftover debugging junk